geth/jail/execution_policy.go

```diff
@@ -71,74 +70,66 @@ func (ep *ExecutionPolicy) executeSendTransaction(req common.RPCCall, call otto.
 	}
 
 	// invoke post processing
-	postProcessRequest(call.Otto, req, messageID)
+	postProcessRequest(vm, req, messageID)
 
 	// @TODO(adam): which one is actually used?
-	res.Set("result", tx.Hash.Hex())
-	res.Set("hash", tx.Hash.Hex())
+	res["result"] = tx.Hash.Hex()
+	res["hash"] = tx.Hash.Hex()
 
 	return res, nil
 }
 
-func (ep *ExecutionPolicy) executeWithClient(client *rpc.Client, req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
-	JSON, err := call.Otto.Object("JSON")
-	if err != nil {
-		return nil, err
-	}
-
+func (ep *ExecutionPolicy) executeWithClient(client *rpc.Client, vm *vm.VM, req common.RPCCall) (map[string]interface{}, error) {
 	var result json.RawMessage
 
-	resp, _ := call.Otto.Object(`({"jsonrpc":"2.0"})`)
-	resp.Set("id", req.ID)
+	resp := map[string]interface{}{
+		"jsonrpc": "2.0",
+		"id":      req.ID,
+	}
 
 	// do extra request pre processing (persist message id)
 	// within function semaphore will be acquired and released,
 	// so that no more than one client (per cell) can enter
-	messageID, err := preProcessRequest(call.Otto, req)
+	messageID, err := preProcessRequest(vm)
 	if err != nil {
 		return nil, common.StopRPCCallError{Err: err}
 	}
 
 	err = client.Call(&result, req.Method, req.Params...)
 	if err != nil {
 		if err2, ok := err.(gethrpc.Error); ok {
-			resp.Set("error", map[string]interface{}{
+			resp["error"] = map[string]interface{}{
 				"code":    err2.ErrorCode(),
 				"message": err2.Error(),
-			})
+			}
 		} else {
-			resp = newErrorResponse(call.Otto, -32603, err.Error(), &req.ID).Object()
+			resp = newErrorResponse(-32603, err.Error(), &req.ID)
```

Comment:
Maybe move this -32603 constant to constant definition, similar to https://github.com/status-im/status-go/blob/develop/geth/rpc/call_raw.go#L13:
```go
const (
	...
	errInvalidMessageCode = -32700 // from go-ethereum/rpc/errors.go
)
```